src/rmp/src/blifParser.cpp

```diff
@@ -0,0 +1,189 @@
+#include "rmp/blifParser.h"
+
+#include <boost/bind.hpp>
+#include <boost/config/warning_disable.hpp>
+#include <boost/fusion/algorithm.hpp>
+#include <boost/fusion/container.hpp>
+#include <boost/fusion/include/adapt_struct.hpp>
+#include <boost/fusion/include/at_c.hpp>
+#include <boost/fusion/include/define_struct.hpp>
+#include <boost/fusion/include/io.hpp>
+#include <boost/fusion/sequence.hpp>
+#include <boost/fusion/sequence/intrinsic/at_c.hpp>
+#include <boost/lambda/lambda.hpp>
+#include <boost/optional/optional_io.hpp>
+#include <boost/spirit/include/phoenix_core.hpp>
+#include <boost/spirit/include/phoenix_operator.hpp>
+#include <boost/spirit/include/qi.hpp>
+#include <boost/spirit/include/qi_alternative.hpp>
+#include <iostream>
+#include <string>
+#include <vector>
+
+namespace blif_parser {
+
+namespace qi = boost::spirit::qi;
+namespace ascii = boost::spirit::ascii;
+namespace phoenix = boost::phoenix;
+
+using boost::fusion::at_c;
+using boost::spirit::ascii::space_type;
+using boost::spirit::ascii::string;
+using boost::spirit::qi::lit;
+using qi::lexeme;
+
+using ascii::char_;
+using ascii::space;
+
+using phoenix::ref;
+using qi::double_;
+using qi::int_;
+
+void setNewInput(std::string input, rmp::BlifParser* parser)
+{
+  if (input != "\\")
+    parser->addInput(input);
+}
+
+void setNewOutput(std::string output, rmp::BlifParser* parser)
+{
+  if (output != "\\")
```

Comment:
There shouldn't be a case where this actually happens, the boost parser doesn't consume empty string and the parser we defined here specifies that there needs to be at least one output, otherwise, the parser will fail.